server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -430,7 +441,143 @@ public Consumer getConsumer(
             this.consumerEnricher.enrich(consumer);
         }
 
-        return consumer;
+        return this.translator.translate(consumer, ConsumerDTO.class);
+    }
+
+    /**
+     * Populates the specified entity with data from the provided DTO. This method will not set the
+     * ID, entitlementStatus, complianceStatusHash, idCert, entitlements, keyPair, canActivate
+     * because clients are not allowed to create or update those properties.
+     *
+     * while autoheal is populated, it is overridden in create method.
+     *
+     * owner is not populated because create and update populate it differently.
+     *
+     * type is not populated because we can specify it during create but not update.
+     *
+     * @param entity
+     *  The entity instance to populate
+     *
+     * @param dto
+     *  The DTO containing the data with which to populate the entity
+     *
+     * @throws IllegalArgumentException
+     *  if either entity or dto are null
+     */
+    protected void populateEntity(Consumer entity, ConsumerDTO dto) {
+        if (entity == null) {
+            throw new IllegalArgumentException("the consumer model entity is null");
+        }
+
+        if (dto == null) {
+            throw new IllegalArgumentException("the consumer dto is null");
+        }
+
+        if (dto.getUuid() != null) {
+            entity.setUuid(dto.getUuid());
+        }
+
+        if (dto.getName() != null) {
+            entity.setName(dto.getName());
+        }
+
+        if (dto.getFacts() != null) {
+            entity.setFacts(dto.getFacts());
+        }
+
+        if (dto.getUsername() != null) {
+            entity.setUsername(dto.getUsername());
+        }
+
+        if (dto.getServiceLevel() != null) {
+            entity.setServiceLevel(dto.getServiceLevel());
+        }
+
+        if (dto.getReleaseVersion() != null) {
+            entity.setReleaseVer(new Release(dto.getReleaseVersion()));
+        }
+
+        if (dto.getEnvironment() != null) {
+            Environment env = environmentCurator.find(dto.getEnvironment().getId());
+            if (env == null) {
+                throw new NotFoundException(i18n.tr("Environment ''{0}'' could not be found.",
+                    dto.getEnvironment().getId()));
+            }
+            entity.setEnvironment(env);
+        }
+
+        if (dto.getLastCheckin() != null) {
+            entity.setLastCheckin(dto.getLastCheckin());
+        }
+
+        if (dto.getCapabilities() != null) {
+            Set<ConsumerCapability> capabilities = new HashSet<ConsumerCapability>();
+            for (CapabilityDTO capabilityDTO : dto.getCapabilities()) {
+                if (capabilityDTO != null) {
+                    capabilities.add(new ConsumerCapability(entity, capabilityDTO.getName()));
+                }
+            }
+            entity.setCapabilities(capabilities);
+        }
```

Comment:
Lot of refactoring here to be done with the encapsulation changes

Reply:
im missing something obvious here as well. will chat with you online